File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Declare the main variables
    int coins = 0;
    float input;
    //Ask how much change is to do, round it to cents and translate it to an integer
    do
    {
        input = get_float("Change owed: ");
    }
    while (input < 0);

    int change = round(input * 100);

    //Set the value of the coins
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    //Check if quarters are needed, how many and go on with the change left
    while (change >= quarter)
    {
        change = change - quarter;
        coins++;
    }

    //Check if dimes are needed, how many and go on with the change left
    while (change >= dime)
    {
        change = change - dime;
        coins++;
    }

    //Check if nickels are needed, how many and go on with the change left
    while (change >= nickel)
    {
        change = change - nickel;
        coins++;
    }

    //Check if pennies are needed, how many and go on with the change left
    while (change >= penny)
    {
        change = change - penny;
        coins++;
    }

    printf("%i\n", coins);

}

File: pset6/cash/cash.py
from cs50 import get_float

# Define the main function


def main():
    # Ask the user for how much change is to do
    change = get_change() * 100

    # Set the money value and the coins counter variable
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    coins = 0

    # Count coins needed recursively starting from bigger to smaller ones
    while (change >= quarter):
        change -= quarter
        coins += 1

    while (change >= dime):
        change -= dime
        coins += 1

    while (change >= nickel):
        change -= nickel
        coins += 1

    while (change >= penny):
        change -= penny
        coins += 1

    # Print the number of coins needed
    print(f"{coins}")


# Define the main function
def get_change():
    while True:
        c = get_float("Change owed: ")
        if c >= 0:
            return c


if __name__ == "__main__":
    main()